services/auth/oauth2.go

```diff
@@ -118,12 +130,12 @@ func (o *OAuth2) Verify(req *http.Request, w http.ResponseWriter, store DataStor
 	}
 
 	id := o.userIDFromToken(req, store)
-	if id <= 0 {
+	if id == -1 || id <= -3 { // -2 means actions, so we need to allow it.
```

Comment:
Hmm, `0` was also a previously disallowed value. Should it be disallowed again or is the new approach "more correct"?
```suggestion
	if id <= 0 && id != -2 { // -2 means actions, so we need to allow it.
```
Looks more clear to me. If `0` is explicitly missing, replace `<=` with `<`.
Also, we should probably extract a constant instead of using magic literals with a comment…